src/material_pool.hpp

```diff
@@ -1,50 +1,53 @@
 #pragma once
 
 #include <string_view>
+#include <unordered_map>
 #include <vector>
 #include <optional>
+#include <stdint.h>
 #include <d3d12.h>
 
+#include "structs.hpp"
 #include "util/defines.hpp"
+#include "id_factory.hpp"
 
 struct aiMaterial;
 
 namespace wr
 {
-	struct TextureHandle {};
-	struct MaterialHandle {};
-	
-	enum class TextureType
-	{
-		PNG,
-		DDS,
-		HDR
-	};
-
 	class MaterialPool
 	{
 	public:
-		explicit MaterialPool(std::size_t size_in_mb);
+		explicit MaterialPool();
 		virtual ~MaterialPool() = default;
 
 		MaterialPool(MaterialPool const &) = delete;
 		MaterialPool& operator=(MaterialPool const &) = delete;
 		MaterialPool(MaterialPool&&) = delete;
 		MaterialPool& operator=(MaterialPool&&) = delete;
 
-		[[nodiscard]] MaterialHandle* Load(std::string_view path, TextureType type);
+		//Creates an empty material. The user is responsible of filling in the texture handles.
+		//TODO: Give Materials default textures
+		[[nodiscard]] MaterialHandle Create();
+		[[nodiscard]] MaterialHandle Create(TextureHandle& albedo,
+											TextureHandle& normal,
+											TextureHandle& rough_met,
+											TextureHandle& ao, 
+											bool is_alpha_masked = false, 
+											bool is_double_sided = false);
+
 		[[nodiscard]] MaterialHandle* Load(aiMaterial* material);
 
-		virtual void Evict() = 0;
-		virtual void MakeResident() = 0;
+		virtual void Evict() {}
```

Comment:
Should be pure virtual.

Reply:
I removed the D3D12MaterialPool since a MaterialPool is enough being platform independent for now